yt/yt/ytlib/api/native/helpers.cpp

```diff
@@ -207,4 +215,25 @@ TError MakeRevivalError(
 
 ////////////////////////////////////////////////////////////////////////////////
 
+void CheckConsumerPermission(
```

Comment:
Let's expose the context switch happening inside by returning a future? We can even chain this with `.Apply`.